Three tests are failing that didn't fail before
```
2021-01-31T17:14:52.7100529Z Failures: 
2021-01-31T17:14:52.7100926Z 	From test_python3.vim:
2021-01-31T17:14:52.7101503Z 	Found errors in Test_Write_To_Current_Buffer_Fixes_Cursor_Clear():
2021-01-31T17:14:52.7106816Z 	Caught exception in Test_Write_To_Current_Buffer_Fixes_Cursor_Clear(): Vim(return):Error invoking 'python_execute' on channel 3 (python3-script-host): Traceback (most recent call last):   File "<string>", line 1, in <module>   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/buffer.py", line 65, in __setitem__     return self.request('nvim_buf_set_lines', start, end, False, lines)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/common.py", line 58, in request     return self._session.request(name, self, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 182, in request     res = self._session.request(name, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/msgpack_rpc/session.py", line 104, in request     raise self.error_wrapper(err) pynvim.api.common.NvimError: Vim:E315: ml_get: invalid lnum: 11 @ function RunTheTest[37]..Test_Write_To_Current_Buffer_Fixes_Cursor_Clear[3]..provider#python3#Call, line 18
2021-01-31T17:14:52.7110779Z 	Found errors in Test_Write_To_Current_Buffer_Fixes_Cursor_List():
2021-01-31T17:14:52.7115885Z 	Caught exception in Test_Write_To_Current_Buffer_Fixes_Cursor_List(): Vim(return):Error invoking 'python_execute' on channel 3 (python3-script-host): Traceback (most recent call last):   File "<string>", line 1, in <module>   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/buffer.py", line 65, in __setitem__     return self.request('nvim_buf_set_lines', start, end, False, lines)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/common.py", line 58, in request     return self._session.request(name, self, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 182, in request     res = self._session.request(name, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/msgpack_rpc/session.py", line 104, in request     raise self.error_wrapper(err) pynvim.api.common.NvimError: Vim:E315: ml_get: invalid lnum: 11 @ function RunTheTest[37]..Test_Write_To_Current_Buffer_Fixes_Cursor_List[3]..provider#python3#Call, line 18
2021-01-31T17:14:52.7119946Z 	Found errors in Test_Write_To_Current_Buffer_Fixes_Cursor_Str():
2021-01-31T17:14:52.7124938Z 	Caught exception in Test_Write_To_Current_Buffer_Fixes_Cursor_Str(): Vim(return):Error invoking 'python_execute' on channel 3 (python3-script-host): Traceback (most recent call last):   File "<string>", line 1, in <module>   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/buffer.py", line 61, in __setitem__     return self.request('nvim_buf_set_lines', i, i + 1, True, lines)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/common.py", line 58, in request     return self._session.request(name, self, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 182, in request     res = self._session.request(name, *args, **kwargs)   File "/home/runner/.local/lib/python3.8/site-packages/pynvim/msgpack_rpc/session.py", line 104, in request     raise self.error_wrapper(err) pynvim.api.common.NvimError: Vim:E315: ml_get: invalid lnum: 11 @ function RunTheTest[37]..Test_Write_To_Current_Buffer_Fixes_Cursor_Str[3]..provider#python3#Call, line 18
```

--- a/src/nvim/api/buffer.c
+++ b/src/nvim/api/buffer.c
@@ -1918,7 +1918,6 @@ static void fix_cursor(linenr_T lo, linenr_T hi, linenr_T extra)
       curwin->w_cursor.lnum += extra;
       check_cursor_col();
     } else if (extra < 0) {
-      curwin->w_cursor.lnum = lo;
       check_cursor();
     } else {
       check_cursor_col();

--- a/test/functional/lua/buffer_updates_spec.lua
+++ b/test/functional/lua/buffer_updates_spec.lua
@@ -225,6 +225,14 @@ describe('lua buffer event callbacks: on_lines', function()
     eq(1, meths.get_var('listener_cursor_line'))
   end)
 
+  it('has valid cursor position while deleting lines', function()
+    meths.buf_set_lines(0, 0, -1, true, { "line_1", "line_2", "line_3", "line_4"})
+    meths.win_set_cursor(0, {2, 0})
+    eq(2, meths.win_get_cursor(0)[1])
+    meths.buf_set_lines(0, 0, -1, true, { "line_1", "line_2", "line_3"})
+    eq(2, meths.win_get_cursor(0)[1])
+  end)
+
   it('does not SEGFAULT when calling win_findbuf in on_detach', function()
 
     exec_lua[[

